### PR TITLE
MXLPA-6: Added pipe client and simple message posting

### PR DESCRIPTION
**Pull request by baiumbg**
Ticket: MXLPA-6

### CODE REVIEW

File: src/MXLPersonalArmoryHook/PipeClient.cpp [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+void PipeClient::connect()
+{
+    m_Pipe = CreateFileW(m_PipeAddress.c_str(), GENERIC_READ | GENERIC_WRITE, 0, NULL, OPEN_EXISTING, 0, NULL);
+}
+
+void PipeClient::send(const std::wstring& message)
+{
+    DWORD len;
+    BOOL success = WriteFile(m_Pipe, message.c_str(), message.size() * sizeof(wchar_t), &len, NULL);
```
Comment (romannick) [inline src/MXLPersonalArmoryHook/PipeClient.cpp]: Variable `success` is not used anywhere. Please remove.

File: MXLPersonalArmoryHook/include/PipeClient.h [hunk 2]
```diff
@@ -0,0 +1,22 @@
+#pragma once
+
+#include <Windows.h>
+#include <string>
+
+#define DEFAULT_PIPE_ADDRESSW L"\\\\.\\pipe\\MXLPersonalArmoryHook"
+#define DEFAULT_PIPE_ADDRESS "\\\\.\\pipe\\MXLPersonalArmoryHook"
```
Comment (romannick) [inline MXLPersonalArmoryHook/include/PipeClient.h:7]: Is `DEFAULT_PIPE_ADDRESS ` used anywhere?

**Review by romannick — CHANGES_REQUESTED** [review]

File: src/MXLPersonalArmoryHook/PipeClient.cpp (hunk 1 above)
Comment (baiumbg) [reply src/MXLPersonalArmoryHook/PipeClient.cpp]: Fixed

File: MXLPersonalArmoryHook/include/PipeClient.h (hunk 2 above)
Comment (baiumbg) [reply MXLPersonalArmoryHook/include/PipeClient.h:7]: Not at the moment, but might be needed in the future. This way is consistent with Microsoft's ANSI/Wide versions of everything involving strings.

File: MXLPersonalArmoryHook/src/DLLMain.cpp [hunk 3]
```diff
@@ -8,16 +8,21 @@ DWORD WINAPI D2Thread(LPVOID lpParam)
 {
     HMODULE mxlpaModule = (HMODULE)lpParam;
 
-    while (!D2CLIENT_GetPlayerUnit())
+    if (!!D2CLIENT_GetPlayerUnit())
     {
-        Sleep(200);
+        D2CLIENT_Print(L"You son of a bitch. I'm in.", 0);
```
Comment (romannick) [inline MXLPersonalArmoryHook/src/DLLMain.cpp:13]: ??? 😄

**Review by romannick — APPROVED** [review]

**Pull request merged**